docs/RFCS/enterprise.md

```diff
@@ -10,44 +10,97 @@ CockroachDB's [business model] describes a set of "enterprise features", which
+    string organization_name = 4;
 }
+
 ```
 
-The latest license will be stored in the [settings table] (TODO(dan): replace
-with a link to the final rfc when merged). Once viewing historical settings is
-supported, it can be used to show historical licensing information. The settings
-table is in the system config keyspace and so is gossiped to every node. Upon
-receiving a gossiped value, the node will decode the protobuf and check both the
-valid_until date and the cluster_id. The result of this check is cached until
-the value of the gossiped licence setting changes. When a SQL statement using an
-enterprise feature is executed, this cached value is consulted.
+The latest license will be stored as a gossiped [cluster
+setting](https://github.com/cockroachdb/cockroach/blob/master/docs/RFCS/settings_table.md).
+Once viewing historical settings is supported, it can be used to show historical
+licensing information. When nodes observe the setting change, the value is
+decoded to update an in-memory copy of the `License` struct.
 
-The settings table is reused here to avoid taking another table ID in the system
-config range, of which we have a very limited number. However, licenses are
-different enough from other settings that we're likely to want to treat them
-differently in the future, so we'll create a separate table when/if we need it.
+ When a SQL statement using an enterprise feature is executed, the time and
+ executing clusterID are checked against the current license to determine if it
+ is valid (the result of this check could potentially be cached until the
+ expiration or next change if or when it becomes a performance concern).
 
 To acquire a license, a user visits the registration page on
 https://cockroachlabs.com. Further details are out of scope.
 
 The user applies a license to a cluster in either of two ways: via a page on the
-admin ui or a supplied SQL command run as the `root` user. The registration
-status is immediately surfaced in the admin ui and available via a new `SHOW
-LICENSE` SQL command. This is also recorded as an event in the eventlog.
+admin ui or `SET CLUSTER LICENSE` SQL command (run as the `root` user). Both
+methods must immediately return an error if the license is invalid (malformed or
+not currently valid for the executing cluster), rather than updating the setting
+and potentially leading to subsequent SQL commands failing (e.g. a nightly
+unattended `BACKUP`).
+
+The current registration status is surfaced admin ui and can be inspected via a
+new `SHOW CLUSTER LICENSE` SQL command.
+
+## Grace Period
+
+Suddenly breaking existing workflows can be very disruptive, and would create a
+poor experience for paying customers. Thus Enterprise licenses may remain usable
+for some grace period beyond their expiration time.
+
+## Licenses Without Expirations
+
+In some situations, we may want to issue licenses that are not time-bound.
+Leaving the expiration unset (0) signifies a license has no expiration.
```

Comment:
Last I checked, @danhhz has a different suggestion here.

As written, this specifies the proto and code checking it be "dumb"  i.e. separate checks along the lines of "if lic says `expires at x`, check if time is x" and "if lic says `for cluster x, y`, check cluster is x or y", etc. 

In offline discussions with @danhhz, he's suggested (as I understand it) that we enumerate the products we offer -- e.g. perpetual license, site license, evaluation license, etc -- as values of License.Type, and handling code switching on that to act accordingly.